File: app/src/main/java/com/thedesignerx/saim/new_message_app/BackGroundTaskLogin.java
package com.thedesignerx.saim.new_message_app;

import android.content.Context;
import android.content.Intent;
import android.os.AsyncTask;
import android.telecom.Call;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.io.UnsupportedEncodingException;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.ProtocolException;
import java.net.URL;
import java.net.URLEncoder;
import java.util.ArrayList;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class BackGroundTaskLogin extends AsyncTask<String, Void, String> {


    Context ctx;

    BackGroundTaskLogin(Context ctx) {
        this.ctx = ctx;
    }
    boolean logdin = false;

    @Override
    protected void onPreExecute() {
        super.onPreExecute();
    }

    @Override
    protected String doInBackground(String... params) {

        String log_url = "http://192.168.1.7:9090/loginapp/login.php";
        String method = params[0];
        if (method.equals("login")) {
            String username = params[1];
            String password = params[2];
            try {
                URL url = new URL(log_url);
                HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
                httpURLConnection.setRequestMethod("POST");
                httpURLConnection.setDoOutput(true);
                httpURLConnection.setDoInput(true);
                httpURLConnection.setUseCaches(false);
                OutputStream outputStream = httpURLConnection.getOutputStream();
                BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(outputStream, "UTF-8"));
                String login_data = URLEncoder.encode("username", "UTF-8") + "=" + URLEncoder.encode(username, "UTF-8") + "&" +
                        URLEncoder.encode("password", "UTF-8") + "=" + URLEncoder.encode(password, "UTF-8");
                bufferedWriter.write(login_data);
                bufferedWriter.flush();
                bufferedWriter.close();
                InputStream inputStream = httpURLConnection.getInputStream();
                BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream, "iso-8859-1"));

                StringBuilder result = new StringBuilder();
                String line = "";


                while ((line = bufferedReader.readLine()) != null) {
                    result.append(line);
                }

                return String.valueOf(result);
            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        return null;
    }

    @Override
    protected void onProgressUpdate(Void... values) {
        super.onProgressUpdate(values);
    }


    @Override
    protected void onPostExecute(String result) {
        Toast.makeText(ctx, result, Toast.LENGTH_LONG).show();
        if(result.equals("Welcome user")){
            logdin = true;
        }
        if (logdin){
            Intent intent = new Intent(ctx, MainActivity.class);
            ctx.startActivity(intent);
        }

    }
}

File: app/src/main/java/com/thedesignerx/saim/new_message_app/BackGroundTaskLogin.kt
//package com.thedesignerx.saim.new_message_app
//
//import android.content.Context
//import android.content.Intent
//import android.os.AsyncTask
//import android.widget.Toast
//import java.io.*
//import java.net.HttpURLConnection
//import java.net.MalformedURLException
//import java.net.URL
//import java.net.URLEncoder
//
//class BackGroundTaskLogin internal constructor(var ctx: Context) : AsyncTask<String?, Void?, String?>() {
//    var logdin = false
//    override fun onPreExecute() {
//        super.onPreExecute()
//    }
//
//    protected override fun doInBackground(vararg params: String): String? {
//        val log_url = "http://192.168.1.7:9090/loginapp/login.php"
//        val method = params[0]
//        if (method == "login") {
//            val username = params[1]
//            val password = params[2]
//            try {
//                val url = URL(log_url)
//                val httpURLConnection = url.openConnection() as HttpURLConnection
//                httpURLConnection.requestMethod = "POST"
//                httpURLConnection.doOutput = true
//                httpURLConnection.doInput = true
//                httpURLConnection.useCaches = false
//                val outputStream = httpURLConnection.outputStream
//                val bufferedWriter = BufferedWriter(OutputStreamWriter(outputStream, "UTF-8"))
//                val login_data = URLEncoder.encode("username", "UTF-8") + "=" + URLEncoder.encode(username, "UTF-8") + "&" +
//                        URLEncoder.encode("password", "UTF-8") + "=" + URLEncoder.encode(password, "UTF-8")
//                bufferedWriter.write(login_data)
//                bufferedWriter.flush()
//                bufferedWriter.close()
//                val inputStream = httpURLConnection.inputStream
//                val bufferedReader = BufferedReader(InputStreamReader(inputStream, "iso-8859-1"))
//                val result = StringBuilder()
//                var line: String? = ""
//                while (bufferedReader.readLine().also { line = it } != null) {
//                    result.append(line)
//                }
//                return result.toString()
//            } catch (e: MalformedURLException) {
//                e.printStackTrace()
//            } catch (e: IOException) {
//                e.printStackTrace()
//            }
//        }
//        return null
//    }
//
//    protected override fun onProgressUpdate(vararg values: Void) {
//        super.onProgressUpdate(*values)
//    }
//
//    override fun onPostExecute(result: String?) {
//        Toast.makeText(ctx, result, Toast.LENGTH_LONG).show()
//        if (result == "Welcome user") {
//            logdin = true
//        }
//        if (logdin) {
//            val intent = Intent(ctx, MainActivity::class.java)
//            ctx.startActivity(intent)
//        }
//    }
//
//}